ethcore/private-tx/src/lib.rs

```diff
@@ -146,13 +148,16 @@ pub struct Receipt {
 
 /// Manager of private transactions
 pub struct Provider {
-	encryptor: Arc<Encryptor>,
-	config: ProviderConfig,
+	encryptor: Box<Encryptor>,
+	validator_accounts: HashSet<Address>,
```

Comment:
`validator_accounts` is only accessed with `contains` method, so there it should be `HashSet`, not `Vec`